src/compiler/config/validate-dev-server.ts

```diff
@@ -3,21 +3,25 @@ import { buildError, isBoolean, isNumber, isString, normalizePath } from '@utils
 import { isAbsolute, join } from 'path';
 import { isOutputTargetWww } from '../output-targets/output-utils';
 
-export const validateDevServer = (config: d.Config, diagnostics: d.Diagnostic[]) => {
+export const validateDevServer = (
+  config: d.UnvalidatedConfig,
+  diagnostics: d.Diagnostic[]
+): d.DevServerConfig | undefined => {
   if ((config.devServer === null || (config.devServer as any)) === false) {
-    return null;
+    return undefined;
   }
 
   const flags = config.flags;
   const devServer = { ...config.devServer };
 
-  if (isString(flags.address)) {
+  if (flags && flags.address && isString(flags.address)) {
```

Comment:
I wonder if it makes sense to update the declaration of `flags` on line 14 to default to an object literal like so:

```diff
- const flags = config.flags;
+ const flags = config.flags || {};
```

That way, we don't need to do checks such as this:
```suggestion
  if (flags.address && isString(flags.address)) {
```

or do optional chaining on every usage of `flags` throughout this function.

Thoughts?